pf4j/src/main/java/org/pf4j/util/Unzip.java

```diff
@@ -78,29 +78,34 @@ public void extract() throws IOException {
         try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(source))) {
             ZipEntry zipEntry;
             while ((zipEntry = zipInputStream.getNextEntry()) != null) {
-                try {
-                    File file = new File(destination, zipEntry.getName());
-
-                    // create intermediary directories - sometimes zip don't add them
-                    File dir = new File(file.getParent());
-                    dir.mkdirs();
-
-                    if (zipEntry.isDirectory()) {
-                        file.mkdirs();
-                    } else {
-                        byte[] buffer = new byte[1024];
-                        int length;
-                        try (FileOutputStream fos = new FileOutputStream(file)) {
-                            while ((length = zipInputStream.read(buffer)) >= 0) {
-                                fos.write(buffer, 0, length);
-                            }
+                File file = new File(destination, zipEntry.getName());
+
+                // create intermediary directories - sometimes zip don't add them
+                File dir = new File(file.getParent());
+
+                mkdirsOrThrow(dir);
+
+                if (zipEntry.isDirectory()) {
+                    mkdirsOrThrow(file);
+                } else {
+                    byte[] buffer = new byte[1024];
+                    int length;
+                    try (FileOutputStream fos = new FileOutputStream(file)) {
+                        while ((length = zipInputStream.read(buffer)) >= 0) {
+                            fos.write(buffer, 0, length);
                         }
                     }
-                } catch (FileNotFoundException e) {
-                    log.error("File '{}' not found", zipEntry.getName());
                 }
             }
         }
     }
 
+    private static void mkdirsOrThrow(File dir) throws IOException {
+        if (!dir.exists()) {
```

Comment:
It's not an issue but I think that it's better to put the both checks on a single line (`if (!dir.exists() && !dir.mkdirs())`)

Reply:
Thanks, I don't know what I was thinking here :smile:

Fixed.